app/Models/Album.php

```diff
@@ -52,11 +52,11 @@ class Album extends Model
      * Get an album using some provided information.
      * If such is not found, a new album will be created using the information.
      */
-    public static function getOrCreate(Artist $artist, ?string $name = null, bool $isCompilation = false): self
+    public static function getOrCreate(Artist $artist, ?string $name = null, ?string $albumartist = null): self
     {
         // If this is a compilation album, its artist must be "Various Artists"
-        if ($isCompilation) {
-            $artist = Artist::getVariousArtist();
+        if ((bool) trim($albumartist)) {
+            $artist = $albumartist;
```

Comment:
if this condition is true, $artist will be a string so the `$artist->id` below will crash with the error:

> Trying to get property 'id' of non-object